torchbenchmark/util/distributed/ddp.py

```diff
@@ -0,0 +1,158 @@
+from datetime import datetime
+from statistics import stdev
+from pathlib import Path
+
+from .trainer import Trainer
+
+from torch.cuda import Event
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.profiler import profile, ProfilerActivity, tensorboard_trace_handler
+
+from torchbenchmark.util.e2emodel import E2EBenchmarkModel
+
+import torch
+import torch.distributed as dist
+
+class DDPTrainer(Trainer):
+    DEFAULT_MEASURE_ITERATIONS = 10
+    def __init__(self, args, model_class, batch_size=None, extra_args=[]):
+        super().__init__(args, model_class, mode="SPMD")
+
+        self.setup()
+
+        # create model instance after Trainer setup, so that
+        # visible devices won't be revised in model constructor
+        model: E2EBenchmarkModel = model_class("train", batch_size, extra_args)
+
+        expected_attrs = ["model", "optimizer", "train_dataloader", "accelerator"]
+        assert all(attr in dir(model) for attr in expected_attrs), (
+            "Missing attributes in the input E2EBenchmarkModel implementation: "
+            f"{[attr for attr in expected_attrs if attr not in dir(model)]}"
+        )
+
+        self.model = model.model
+        self.optimizer = model.optimizer
+        self.dataloader = model.train_dataloader
+        self.accelerator = model.accelerator
+
+        self.rank = dist.get_rank()
+        self.world_size = dist.get_world_size()
+
+
+        self.ddp_model = DDP(
+            self.model,
+            # After setting CUDA_VISIBLE_DEVICES to a single device, each
+            # process only sees cuda:0
+            device_ids=[0],
+            # If buffer broadcast is necessary, specific optimizations might be
+            # necessary to optimize performance. Disable it by default.
+            broadcast_buffers=False,
+            # Set gradient as bucket view to avoid unnecessary copies
+            gradient_as_bucket_view=True,
```

Comment:
Might as well set static_graph=True, because if there were no errors, means there were no unused parameters and the graph should be static?